### DISCUSSION TITLE
BusinessRuleAsync and ProvideTargetWhenAsync

### DISCUSSION THREAD

**Discussion by mikeclimbs23** [MarimerLLC/csla#2307] [Questions]
I'm using CSLA 5.41 with blazor and I'm starting to add some rules. One of my rules needs the object type and makes a SQL call.

I'm pulling the object type from the target, but since I'm making a SQL call within the rule I would prefer the rule be async. 

Am I making a big mistake using an async rule and turning on ProvideTargetWhenAsync?

Thanks,

Mike

**Comment by rockfordlhotka** [comment]
Two things:

1. If all you are doing is getting the type of the target, then that's safe because that value can't be changed by a different thread
2. You can't make a SQL call directly from within a rule, because rules might be running on the logical client; you need to invoke the data portal to ask an object to make the SQL call on your behalf (often a readonlybase or commandbase subclass)

**Reply by mikeclimbs23** [reply]
Thank You, very helpful advice.

Mike
